### ISSUE TITLE
Compiling with windows visual studio 2015 will produce error when this flags --input-type-color --input-type-date --input-type-datetime and others used

### ISSUE THREAD

**Issue by dexter4life** [qtwebkit/qtwebkit#937]
> DerivedSources\JavaScriptCore\JSReplayInputs.h(76): warning C4002: too many actual parameters for macro 'GetCurrentTime'
> DerivedSources\JavaScriptCore\JSReplayInputs.h(76): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
> DerivedSources\JavaScriptCore\JSReplayInputs.h(76): warning C4183: 'GetTickCount': missing return type; assumed to be a member function returning 'int'
> DerivedSources\JavaScriptCore\JSReplayInputs.h(77): error C2523: 'JSC::GetCurrentTime::~GetTickCount': destructor tag mismatch

I noticed some point that the reason for that error above is because Microsoft windows API has a macro definition with the name GetCurrentTime and if expanded will be like

`#define GetTickCount() GetCurrentTime() `

So I used this 
```
#ifdef GetCurrentTime
#undef GetCurrentTime
#endif
```

To add it into the code for CodeGeneratorReplayInputsTemplates.py

I will like to know what you guys think.

Thanks

**Comment by annulen** [comment]
Could you please specifiy your full build configuration? Note that enabling input types which are not on by default won't magically make them working without implementing supporting code.

**Comment by dexter4life** [comment]
okay, I am using qtwebkit-5.212.0-alpha2 but it does also behave same way in qtwebkit-5.212.0-alpha3 and I can see that qtwebkit-5.212.0-alpha3 is the latest

**Comment by annulen** [comment]
You can use newer binary builds from http://download.qt.io/snapshots/ci/qtwebkit/5.212/1571922148/qtwebkit/ which are built against Qt 5.13

**Comment by dexter4life** [comment]
So this will get rid of the error I get?

**Comment by dexter4life** [comment]
I want to build it statically, I don't need DLL dependencies.

**Comment by annulen** [comment]
I guess if you won't be building QtWebKit yourself you shouldn't get compilation errors from it :)

**Comment by annulen** [comment]
Ah ok. Patches are welcome

**Comment by dexter4life** [comment]
Yes, I am building it myself, that's why. well thanks, I also want to contribute to this project

**Comment by annulen** [comment]
Sorry that wasn't mean to discourage you. 

Do you have `--web-replay` in build-webkit arguments when getting this error?

**Comment by annulen** [comment]
FWIW, by "full build configuration" I've meant the full list of options, or your CMakeCache.txt

**Comment by dexter4life** [comment]
okay

**Comment by dexter4life** [comment]
Yes i have --web-replay

**Comment by dexter4life** [comment]
[CMakeCache.txt](https://github.com/qtwebkit/qtwebkit/files/4155372/CMakeCache.txt)
So this is my build Cache

**Comment by dexter4life** [comment]
So what does the --web-replay switch does, I am assuming it has to do with videos?

**Comment by annulen** [comment]
This option is not supported. You should use default configuration, unless you want to disable certain features, mentioned in https://github.com/qtwebkit/qtwebkit/wiki/Building-static-libraries. Disabling anything else is not recommended unless you fully understand consequences

If you open your build directory cmake-gui or ccmake you can see a limited list of public options, which are relatively safe to play with. Anything else is private and should not be touched unless you fully understand consequences

**Comment by dexter4life** [comment]
okay, build i am ready for the risk to learn from it.

**Comment by dexter4life** [comment]
According to firefox MDN, 

> Replay allows Firefox content processes to record their behavior, replay it later, and rewind to earlier states. 

So I am assuming the same is the case for this option?

**Comment by annulen** [comment]
In your CMakeCache.txt I see more options which are just not going to work, like input speech.

**Comment by dexter4life** [comment]
okay, I suspected that one before, anyways. thanks, I get rid of it.  But still, I was hesitant, so I can learn and contribute

**Comment by annulen** [comment]
>So I am assuming the same is the case for this option?

Kind of, except it's probably buggy or at least not fully conformant to spec, and was never tested with Qt port. If any Qt-specific code is needed for this feature it's surely not written.

**Comment by dexter4life** [comment]
Yes

**Comment by dexter4life** [comment]
I will play with the options and see what works and why.  Thanks

**Comment by dexter4life** [comment]
My goal is to score 500+ using https://html5test.com/ as a test too.

**Comment by dexter4life** [comment]
So far the score is just 326 or so.

**Comment by annulen** [comment]
In this case you should better experiment qtwebkit-dev-wip branch which is based on much newer WebKit sources.

Rule of thumb is that you should not enable any web-facing features which are OFF by default, because Apple developers don't consider them to be ready for use in production.

**Comment by annulen** [comment]
Enabling non-conformant feature may sometimes add a few extra points, but might break web pages which work fine when feature is disabled. There was a well-known incident with Fetch API and GitHub's polyfill, which stopped being used when Fetch got enabled by default in Safari preview where it was in unfinished state, breaking pages which relid on conformant behavior provided by polyfill or natively.

**Comment by dexter4life** [comment]
okay, thanks

**Comment by dexter4life** [comment]
How can I get the link to this qtwebkit-dev-wip?

**Comment by annulen** [comment]
it's default branch on GitHub :)
